### PR TITLE
Fix STS parsing and handling of unchecked-TLS connections

### PR DESCRIPTION
**Pull request by progval**
thx to @SadieCat for reporting the issue

### CODE REVIEW

File: test/test_irclib.py [hunk 1]
```diff
@@ -840,79 +840,85 @@ def setUp(self):
...
-            args=('*', 'LS', 'sts=duration=42,port=12345')))
+            args=('*', 'LS', 'sts=' + cap_value)))
 
         self.assertEqual(ircdb.networks.getNetwork('test').stsPolicies, {
-            'irc.test': (6697, 'duration=42,port=12345')})
+            'irc.test': (6697, cap_value)})
         self.irc.driver.reconnect.assert_not_called()
 
-    def testStsInSecureConnectionNoPort(self):
+    def testStsInSecureConnectionWithoutPort(self):
```
Comment (jlu5) [inline test/test_irclib.py:854]:
```suggestion
    def testStsInSecureConnectionWithPort(self):
```

File: test/test_irclib.py [hunk 2]
```diff
@@ -840,79 +840,85 @@ def setUp(self):
...
             wait=True)
 
-    def testStsInCleartextConnection(self):
-        self.irc.driver.anyCertValidationEnabled.return_value = False
-        self.irc.driver.ssl = True
-        self.irc.driver.currentServer = drivers.Server('irc.test', 6667, None, False)
-        self.irc.feedMsg(ircmsgs.IrcMsg(command='CAP',
-            args=('*', 'LS', 'sts=duration=42,port=6697')))
+    def testStsInInsecureTlsConnectionWithPort(self):
+        self._testStsInInsecureTlsConnection('duration=42,port=6697')
```
Comment (jlu5) [inline test/test_irclib.py:885]: Out of curiosity, if we're currently on an insecure TLS connection on port 6697, and get e.g. port 12345, should we reconnect on port 12345?

**Review by jlu5 — COMMENTED** [review]
Do we want tests for the `duration=0`, duration update and (ignored) CAP DEL cases? (Up to you)

File: test/test_irclib.py (hunk 1 above)
Comment (jlu5) [reply test/test_irclib.py:854]: (The inline diff doesn't show it well but there were two functions named `testStsInSecureConnectionWithoutPort`)

File: test/test_irclib.py (hunk 2 above)
Comment (progval) [reply test/test_irclib.py:885]: Unspecced, but I'd say yes.

**Pull request merged**